docs/sphinx/developer/components/pipeline/units.rst

```diff
@@ -38,6 +38,31 @@ This process of pre-decoding, predicting, and updating the PC continues until on
   The fetched memory block is exhausted
     The next block may be requested, and processing will resume once the data is available.
 
+.. _loopBuf:
+
+Loop Buffer
+***********
+
+Within the fetch unit is a loop buffer that can store a configurable number of Macro-Ops. The loop buffer can be pulled from instead of memory if a loop is detected. This avoids the need to re-request data from memory if a branch is taken and increases the throughput of the fetch unit.
+
+Each entry of the loop buffer is the encoding of the Macro-Op, therefore, when supplying an instruction from the loop buffer, the pre-decoding step must still be performed. This was required to avoid any issues with multiple instantiations of the same instruction editing eachothers class members.
```

Comment:
"Each entry of the loop buffer is the encoding of the Macro-Op, therefore, when supplying..." -> "Each entry of the loop buffer is the encoding of the Macro-Op. Therefore, when supplying..."